### PR TITLE
feat!: Change default mappings for CLR type decimal

### PR DESCRIPTION
**Pull request by amanda-tarafa**

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
@jskeet Review only the last commit, all the others are from #12557

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs [hunk 1]
```diff
@@ -312,17 +312,17 @@ public void ToString_()
...
               new Key(new SpannerParameterCollection { new SpannerParameter { ParameterName = "k4", Value = new DateTime(2024, 06, 08) } }));
 
-            Assert.Equal("[[ 3.14 ], [ 1.71 ])", range1.ToString(builder));
+            Assert.Equal("[[ \"3.14\" ], [ \"1.71\" ])", range1.ToString(builder));
             Assert.Equal("[[ \"2022-06-08T00:00:00Z\" ], [ \"2024-06-08T00:00:00Z\" ]]", range2.ToString(builder));
 
-            // Specify the type mappings.(Numeric for decimal, Date for DateTime)
-            builder.ClrToSpannerTypeDefaultMappings = "DecimalToNumeric,DateTimeToDate";
-            Assert.Equal("[[ \"3.14\" ], [ \"1.71\" ])", range1.ToString(builder));
+            // Specify the type mappings.(double for decimal, Date for DateTime)
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs:318]: Can we add a space before the `(` while we're here?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs [hunk 2]
```diff
@@ -427,15 +427,15 @@ public void ToStringKeySetFromKeysWithBuilder()
             var keys = KeySet.FromKeys(new Key(new SpannerParameterCollection { new SpannerParameter { ParameterName = "k1", Value = 3.14m } }),
                 new Key(new SpannerParameterCollection { new SpannerParameter { ParameterName = "k2", Value = new DateTime(2022, 06, 08) } }));
 
-            Assert.Equal("KeySet {Keys = [[ 3.14 ], [ \"2022-06-08T00:00:00Z\" ]]}", keys.ToString(builder));
+            Assert.Equal("KeySet {Keys = [[ \"3.14\" ], [ \"2022-06-08T00:00:00Z\" ]]}", keys.ToString(builder));
 
-            // Specify the type mappings.(Numeric for decimal, Date for DateTime)
-            builder.ClrToSpannerTypeDefaultMappings = "DecimalToNumeric,DateTimeToDate";
-            Assert.Equal("KeySet {Keys = [[ \"3.14\" ], [ \"2022-06-08\" ]]}", keys.ToString(builder));
+            // Specify the type mappings.(double for decimal, Date for DateTime)
```
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs:432]: Space again.

**Review by jskeet — APPROVED** [review]
Two requests to add whitespace, but that's all :)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs:318]: Done

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs (hunk 2 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/KeysTests.cs:432]: Done

**Review by amanda-tarafa — COMMENTED** [review]
Spaces added. I'll push shortly and merge on green. Thanks!

**Pull request merged**